src/presentation/http/router/join.test.ts

```diff
@@ -0,0 +1,70 @@
+import { describe, test, expect } from 'vitest';
+
+describe('Join API', () => {
+  describe('POST /join/:hash', () => {
+    test('Returns 406 when user is already in the team', async () => {
+      const hash = 'Hzh2hy4igf';
+      const userId = 2;
+      const accessToken = global.auth(userId);
+
+      const response = await global.api?.fakeRequest({
+        method: 'POST',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/join/${hash}`,
+      });
+
+      expect(response?.statusCode).toBe(406);
+
+      expect(await response?.json()).toStrictEqual({
+        message: 'User already in team',
+      });
+    });
+    test('Returns 406 when invitation hash is not valid', async () => {
+      const hash = 'Jih23y4igf';
+      const userId = 4;
+      const accessToken = global.auth(userId);
+
+      const response = await global.api?.fakeRequest({
+        method: 'POST',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/join/${hash}`,
+      });
+
+      expect(response?.statusCode).toBe(406);
+
+      expect(await response?.json()).toStrictEqual({
+        message: `Wrong invitation`,
+      });
+    });
+
+    test('Returns 200 when user is added to the team', async () => {
+      const userId = 1;
+      const noteId = 1;
```

Comment:
i guess there is no need to separate noteId if you use it one time

Reply:
done